server/routes/nl_interface_next.py

```diff
@@ -518,63 +246,50 @@ def _detection(orig_query, cleaned_query, embeddings_build,
   if contained_in_classification is not None:
     classifications.append(contained_in_classification)
 
-    # Check if the contained in referred to COUNTRY type. If so,
-    # and the default location was chosen, then set it to Earth.
-    if (place_detection.using_default_place and
-        (contained_in_classification.attributes.contained_in_place_type
-         == ContainedInPlaceType.COUNTRY)):
-      logging.info(
-          "Changing detected place to Earth because no place was detected and contained in is about countries."
-      )
-      place_detection.main_place.dcid = "Earth"
-      place_detection.main_place.name = "Earth"
-      place_detection.main_place.place_type = "Place"
-      place_detection.using_default_place = False
-
   # Correlation classification
   correlation_classification = model.heuristic_correlation_classification(query)
   logging.info(f'Correlation classification: {correlation_classification}')
   if correlation_classification is not None:
     classifications.append(correlation_classification)
 
-  # Clustering-based different SV detection is only enabled in LOCAL.
-  if os.environ.get('FLASK_ENV') == 'local' and svs_scores_dict:
-    # Embeddings Indices.
-    sv_index_sorted = []
-    if 'EmbeddingIndex' in svs_scores_dict:
-      sv_index_sorted = svs_scores_dict['EmbeddingIndex']
-
-    # Clustering classification, currently disabled.
-    # clustering_classification = model.query_clustering_detection(
-    #     embeddings_build, query, svs_scores_dict['SV'],
-    #     svs_scores_dict['CosineScore'], sv_index_sorted,
-    #     COSINE_SIMILARITY_CUTOFF)
-    # logging.info(f'Clustering classification: {clustering_classification}')
-    # logging.info(f'Clustering Classification is currently disabled.')
-    # if clustering_classification is not None:
-    #   classifications.append(clustering_classification)
-
   if not classifications:
-    # Simple Classification simply means:
-    # Use the main place and matched SVs. There are no
-    # rankings, temporal, contained_in or correlations.
+    # if not classification is found, it should default to UNKNOWN (not SIMPLE)
     classifications.append(
-        NLClassifier(type=ClassificationType.SIMPLE,
+        NLClassifier(type=ClassificationType.UNKNOWN,
                      attributes=SimpleClassificationAttributes()))
 
   return Detection(original_query=orig_query,
                    cleaned_query=cleaned_query,
                    places_detected=place_detection,
                    svs_detected=sv_detection,
+                   query_type=_query_type_from_classifications(classifications),
                    classifications=classifications)
 
 
+def _query_type_from_classifications(classifications):
+  ans = ClassificationType.SIMPLE
+  for cl in classifications:
+    if (_classification_to_int(cl.type) > _classification_to_int(ans)):
+      ans = cl.type
+  return ans
+
+
+def _classification_to_int(en):
```

Comment:
i'm quite puzzled by this (the numbers don't match the classificationtype enum values). can you elaborate more with comments, and is there a more robust way to specify the conversion values? also, why is 2 missing